src/main/java/com/spartronics4915/frc2020/commands/IndexerCommands.java

```diff
@@ -229,12 +230,25 @@ public boolean isFinished()
         public LoadToLauncher(Indexer indexer, int ballsToShoot)
         {
             mIndexer = indexer;
+            double spinDistance = (mIndexer.getSlotBallLoaded() && mIndexer.getIntakeBallLoaded()) ? 0.5 : 0;
 
             addCommands(
+                new Spin(mIndexer, -spinDistance),
                 new StartLaunch(mIndexer),
-                new LoadBallToSlot(mIndexer, ballsToShoot),
+                new LoadBallToSlot(mIndexer, ballsToShoot + spinDistance),
                 new EndLaunch(mIndexer)
             );
         }
+
+        public LoadToLauncher(Indexer indexer)
+        {
+            this(indexer, 5);
+        }
+
+        @Override
+        public void end(boolean interrupted)
+        {
+            mIndexer.toNearestQuarterRotation();
```

Comment:
two comments here: 

1) Indexer might want to keep a member variable for the current target position. And probably should post that value to the smartdashboard for debugging.

2) The end methods above seem to be operating on the idea that the command can be expected to complete.  It's not obvious to this reader, but I would anticipate that having requested this (and ending) that there must be a default command for the subsystem that continuously establishes the set point.  The fact that there is no member variable in indexer for the current set point make me wonder what is expected to be going on here.

Reply:
So I would recommend that you add a DefaultCommand into IndexerCommands.java.  Then make sure that that's identified as the default, via this.setDefaultCommand.   I see that  RobotContainer is doing some of this at around line 103.  I would suggest that this is the subsystem's responsibility and it would be better for it to happen in subsystem initialization.

Reply:
perhaps @j-james can be convinced to move this stuff out of RobotContainer.